src/net_processing.cpp

```diff
@@ -4155,7 +4154,7 @@ bool PeerManager::SendMessages(CNode* pto)
             // Only actively request headers from a single peer, unless we're close to today.
             if ((nSyncStarted == 0 && fFetch) || pindexBestHeader->GetBlockTime() > GetAdjustedTime() - 24 * 60 * 60) {
                 state.fSyncStarted = true;
-                state.nHeadersSyncTimeout = GetTimeMicros() + HEADERS_DOWNLOAD_TIMEOUT_BASE + HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * (GetAdjustedTime() - pindexBestHeader->GetBlockTime())/(consensusParams.nPowTargetSpacing);
+                state.nHeadersSyncTimeout = current_time.count() + HEADERS_DOWNLOAD_TIMEOUT_BASE + HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER * (GetAdjustedTime() - pindexBestHeader->GetBlockTime())/(consensusParams.nPowTargetSpacing);
```

Comment:
This is not type safe, because someone could changed the type of current_time to have more or less precision than it has now. Please see the `count_` helpers.

Reply:
That's indeed better, done.

A bit more abstractly, I don't think that's very different from the risk of `nHeadersSyncTimeout` being changed to represent a time in nanoseconds instead of microseconds, for example. We're mixing std::chrono types with integers, and the conversions will always involve these risks. The correct approach is to get rid of all of those, and replace them all with `std::chrono` types. I'm happy to look into that as follow-up PR.